addons/mail/static/src/core/common/thread_model.js

```diff
@@ -657,12 +657,12 @@ export class Thread extends Record {
         try {
             const { data, messages } = await this.fetchMessagesData({ after, around, before });
             this.store.insert(data, { html: true });
-            this.isLoaded = true;
             return this.store.Message.insert(messages.reverse());
         } catch (e) {
             this.hasLoadingFailed = true;
             throw e;
         } finally {
+            this.isLoaded = true;
```

Comment:
Looks ok too. I remember at some point we meant `isLoaded` as "successfully loaded".
When looking at the current state of code, `isLoaded` is used for end of loading so that's good, and all configurations seem robust with broadening the definition like this.